example/CMakeLists.txt

```
@@ -12,7 +12,7 @@ endif()
 # Example of using pre-compiled library
 # ---------------------------------------------------------------------------------------
 add_executable(example example.cpp)
-target_link_libraries(example PRIVATE spdlog::spdlog)
+target_link_libraries(example PRIVATE spdlog::spdlog $<$<BOOL:${MINGW}>:ws2_32>)
```

```suggestion
target_link_libraries(example PRIVATE spdlog::spdlog $<$<PLATFORM_ID:MinGW>:ws2_32>)
```

You can also use CMake generator-expression `$<PLATFORM_ID:MinGW>`. This is just a suggestion since the result is the same.

---

Thanks! I tried it, but it didn't work. Maybe because of the names mentioned in [CMake](https://github.com/Kitware/CMake/blob/323f5feb93a764bf46698958b871adc3e76b2270/Modules/CMakeDetermineSystem.cmake#L5-L31) and in the [documentation](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#platform).

I found out that there's a shorter version:
```cmake
target_link_libraries(example PRIVATE spdlog::spdlog $<${MINGW}>:ws2_32>)
```

---

It is odd that it does not work since it has to be defined in the CMake v3 source.

https://github.com/Kitware/CMake/blob/01a25d72c69b0a9a60c1a62939d6a2c543448118/Modules/CMakePlatformId.h.in#L14-L15

Well, the result is the same, so it is fine as it is now.
I just thought it would be better to unify with CMake generator-expression.

---

Yeah, strange. I'm using cmake version 3.24.2.